sendQueued.js

```diff
@@ -1,130 +1,158 @@
+ * @param  {string} from who the message is being sent from
+ * @param  {string} body message to be sent
+ * @return {Promise} Promise to send message.
+ */
+function sendMessage(to, from, body) {
+  return chainable(function(resolve, reject) {
+    client.sendMessage({to: to, from: from, body: body}, genericResolver(resolve, "client.sendMessage"));
   });
-}
+};
 
-var cleanupName = function(name) {
-  // Switch LAST, FIRST to FIRST LAST
-  // var bits = name.split(',');
-  // name = bits[1] + ' ' + bits[0];  // Alaska already has this format
-  name = name.trim();
+/**
+ * Update queued message in db to indicate it has been sent, and that a reminder will be sent.
+ * 
+ * @param  {string} queuedId index by which to lookup queued message for update.    
+ * @return {Promise} function to recieve results and Promise to perform update.
+ */
+function updateSentWithReminder(queuedId) {
+    return chainable(function(resolve, reject) {
+        knex('queued')
+          .where('queued_id', '=', queuedId)
+          .update({'sent': true,
+                    'asked_reminder': true,
+                    'asked_reminder_at' : dates.now()})
+          .asCallback(genericResolver(resolve, "updateSentWithReminder()"));
+    });
+};
 
-  // Change FIRST LAST to First Last
-  name = name.replace(/\w\S*/g, function(txt) { return txt.charAt(0).toUpperCase() + txt.substr(1).toLowerCase(); });
+/**
+ * Update data for queued message to indicate it has been sent but no reminder is required.
+ * 
+ * @param  {string} queuedId index to be used for lookup of queued message when updating.
+ * @return {function} function to recieve results and Promise to perform update.
+ */
+function updateSentWithoutReminder(queuedId) {
+    return chainable(function(resolve, reject) {
+        knex('queued')
+          .where('queued_id', '=', queuedId)
+          .update({'sent': true})
+          .asCallback(genericResolver(resolve, "updateSentWithoutReminder()"));
+    });
+};
 
-  return name;
+/**
+ * Generic callback handler for resolving a promise once a call has completed.
+ * 
+ * @param  {function} resolve resolve function for Promise that is to be resolved.
+ * @param  {string} errPrefix String prefix for error message if call fails and an error is returned.
+ */
+function genericResolver(resolve, errPrefix) {
+  return function(err, result) {
+    if (err) {
+      return console.log(errPrefix, err);
+    }
+    resolve(result);
+  };
 };
 
+/**
+ * Hook for processing all applicable queued messages.
+ * 
+ * @return {Promise} Promise to process all queued messages.
+ */
 module.exports = function() {
   return new Promise(function(resolve, reject) {
-    findQueued().then(function(resp) {
-      sendQueuedMessage(resp).then(resolve, reject);
-    }).catch(reject);
+    try{
+      findQueued()
+        .then(forEachResult(retrieveCitation))
+        .then(forEachResult(processCitationMessage))
+        .then(resolve, reject);
+    } catch(e) {
```

Comment:
I'm a promise novice, but won't this promise pass errors to `.catch()` instead of throwing them?